src/defs.js

```diff
@@ -17,20 +22,30 @@ if (typeof Uint8Array.from === "function") {
     };
 }
 
+/**
+ * Returns the index of the first occurrence of the character in the Lua string.
+ *
+ * @type {function(Uint8Array|any[], number, number=):number}
+ */
 let luastring_indexOf;
 if (typeof (new Uint8Array().indexOf) === "function") {
     luastring_indexOf = function(s, v, i) {
         return s.indexOf(v, i);
     };
 } else {
     /* Browsers that don't support Uint8Array.indexOf seem to allow using Array.indexOf on Uint8Array objects e.g. IE11 */
-    let array_indexOf = [].indexOf;
+    let array_indexOf = [0].indexOf;
```

Comment:
This is actually because TypeScript treats `[]` as `Array<undefined>` and forbids using it as `Array<number>`. Probably using `Array.prototype.indexOf` is more understandable.